File: solutions/julia/pascals-triangle/pascals-triangle.jl
function triangle(n)
    if n < 0
        throw(DomainError(n))
    end
    if n == 0
        return []
    end
    r = triangle(n - 1)
    if isempty(r)
      return [[1]]
    end
    l = r[end]
    return [
      r...,
      [0, l...] + [l..., 0]
    ]
end

File: solutions/dart/pascals-triangle/lib/pascals_triangle.dart
class PascalsTriangle {
  rows(int n) {
    if (n < 1) {
      return <List<int>>[];
    } else {
      var r = this.rows(n - 1);
      var last = r.isEmpty ? [] : r[r.length - 1];
      return [...r,
        [1,
          for (var i = 1; i <= last.length; ++i)
           (i == last.length ? 0 : last[i]) + last[i - 1]
      ]];
    }
  }
}
